### ISSUE TITLE
There's no way to add a colorbar

### ISSUE THREAD

**Issue by tkarna** [jupyter-widgets/ipyleaflet#706]
As far as I can tell, currently there's no way to add a colorbar or legend to layers. Without a legend, the visualization is not meaningful.

**Comment by martinRenou** [comment]
ipyleaflet could indeed have this built-in.

Although you can put anything you want in a `WidgetControl`, if you find a library that makes nice colorbars for you (branca?) simply put this colorbar in an `Output` widget and put it in a `WidgetControl` :)

**Comment by davidbrochart** [comment]
Linking with https://github.com/davidbrochart/xarray_leaflet/issues/14, in case you want to use xarray-leaflet.

**Comment by giswqs** [comment]
Yes, this would be nice to add https://github.com/giswqs/geemap/issues/223

**Comment by giswqs** [comment]
I have implemented the colorbar in [geemap](https://geemap.org/). 

Notebook: https://geemap.org/notebooks/55_raster_vis
Source code:https://github.com/giswqs/geemap/blob/master/geemap/geemap.py#L2382

![](https://i.imgur.com/XcL0tih.gif)

**Comment by jcy204** [comment]
You can also use a bqplot and just hide the content:

```python
from bqplot import Figure, ColorAxis, ColorScale 

c_sc = ColorScale(colors=['red','green'],animation_duration=500)

ax_c = ColorAxis(scale=c_sc,
                 label_offset='1000px',
                 orientation='horizontal',
                 side='bottom')

map_cbar = Figure(axes=[ax_c],
                  fig_margin=dict(top=0, bottom=-80, left=-80, right=-80),
                  layout={
                      'height': '60px',
                      'width': '300px'
                  })
map_cbar

map_color_widget = WidgetControl(widget=map_cbar, position='bottomright')
m.add_control(map_color_widget)

c_sc.min=100
c_sc.max=1000
```

**Comment by giswqs** [comment]
Here are two examples for creating collarbars and colormaps in leafmap with a few lines of code. 

https://leafmap.org/notebooks/07_colorbar

https://leafmap.org/notebooks/23_colormaps

**Comment by HaudinFlorence** [comment]
Related to this issue, recently a `ColormapControl`has been added to ipyleaflet https://github.com/jupyter-widgets/ipyleaflet/pull/1010. It can be used with layers displaying data with a color mapping (Choropleth, Velocity, Heatmap).
It is built as a widget control and relies on Branca Colormap. Some changes have been proposed to customize it (adapt its size, add a caption, add label ticks) (https://github.com/python-visualization/branca/pull/110)
